apps/studio/components/interfaces/Organization/Usage/Usage.constants.tsx

```diff
@@ -100,73 +100,93 @@ export const USAGE_CATEGORIES: (subscription?: OrgSubscription) => CategoryMeta[
+              const usageRatio =
+                typeof usageMeta !== 'number'
+                  ? (usageMeta?.usage ?? 0) / (usageMeta?.pricing_free_units ?? 0)
+                  : 0
+              const hasLimit = usageMeta && (usageMeta?.pricing_free_units ?? 0) > 0
 
-          const isApproachingLimit = hasLimit && usageRatio >= USAGE_APPROACHING_THRESHOLD
-          const isExceededLimit = hasLimit && usageRatio >= 1
-          const isCapped = usageMeta?.capped
+              const isApproachingLimit = hasLimit && usageRatio >= USAGE_APPROACHING_THRESHOLD
+              const isExceededLimit = hasLimit && usageRatio >= 1
+              const isCapped = usageMeta?.capped
 
-          const onFreePlan = subscription?.plan?.name === 'Free'
+              const onFreePlan = subscription?.plan?.name === 'Free'
 
-          return (
-            <div>
-              {(isApproachingLimit || isExceededLimit) && isCapped && (
-                <Alert
-                  withIcon
-                  variant={isExceededLimit ? 'danger' : 'warning'}
-                  title={
-                    isExceededLimit
-                      ? 'Exceeding database size limit'
-                      : 'Nearing database size limit'
-                  }
-                >
-                  <div className="flex w-full items-center flex-col justify-center space-y-2 md:flex-row md:justify-between">
-                    <div>
-                      When you reach your database size limit, your project can go into read-only
-                      mode.{' '}
-                      {onFreePlan
-                        ? 'Please upgrade your Plan.'
-                        : 'Disable your spend cap to scale seamlessly and pay for over-usage beyond your Plans quota.'}
-                    </div>
-                  </div>
-                </Alert>
-              )}
-            </div>
-          )
-        },
-      },
+              return (
+                <div>
+                  {(isApproachingLimit || isExceededLimit) && isCapped && (
+                    <Alert
+                      withIcon
+                      variant={isExceededLimit ? 'danger' : 'warning'}
+                      title={
+                        isExceededLimit
+                          ? 'Exceeding database size limit'
+                          : 'Nearing database size limit'
+                      }
+                    >
+                      <div className="flex w-full items-center flex-col justify-center space-y-2 md:flex-row md:justify-between">
+                        <div>
+                          When you reach your database size limit, your project can go into
+                          read-only mode.{' '}
+                          {onFreePlan
+                            ? 'Please upgrade your Plan.'
+                            : 'Disable your spend cap to scale seamlessly and pay for over-usage beyond your Plans quota.'}
```

Comment:
I'll do my best Terry impression...
```suggestion
                            : 'Disable your spend cap to scale seamlessly, and pay for over-usage beyond your Plan's quota.'}
```